Sources/Instrumentation/Instrument.swift

```diff
@@ -1,50 +1,58 @@
 import Baggage
 
-/// Conforming types are usually cross-cutting tools like tracers that extract values of one type (`ExtractFrom`) into a
-/// `BaggageContext` and inject values stored in a `BaggageContext` into another type (`InjectInto`).
-/// `ExtractFrom` and `InjectInto` may very well be of the same type, e.g. when a cross-cutting tool propagates
-/// baggage through HTTP headers.
-public protocol InstrumentProtocol {
-    associatedtype InjectInto
-    associatedtype ExtractFrom
+/// Conforming types are used to carry metadata over a boundary (e.g. HTTP).
+public protocol CarrierProtocol {}
```

Comment:
I'm not sure we need this and it may have negative compatibility implications ... 
I need to sanity check if I remember this right though, will loop back soon

Reply:
Ah okey the thing I remembered is "just" https://github.com/apple/swift-nio/blob/master/docs/public-api.md#2-conforming-nio-types-to-protocols 

which is a bit of a mix for us and mostly okey... since "we" the tracing impls providing the conformance would do this.

Having that said, I don't think we need this protocol -- we can express everything with "just" `associatedtype Carrier` without the `: CarrierProtocol`. let's remove to keep the API minimal?

Reply:
Yes, let's remove it 👌